bin/tools.py

```diff
@@ -861,6 +876,36 @@ def run_parsed_arguments(args):
         sys.exit(1)
 
 
+def read_personal_config():
+    args = {}
+
+    if is_windows():
+        home_config = Path(os.getenv('AppData'))
+    else:
+        home_config = (
+            Path(os.getenv('XDG_CONFIG_HOME'))
+            if os.getenv('XDG_CONFIG_HOME')
+            else Path(os.getenv('HOME')) / '.config'
+        )
+
+    for config_file in [
+        # Highest prio: contest directory
+        Path() / '.bapctools.yaml',
+        Path() / '..' / '.bapctools.yaml',
+        # Lowest prio: user config directory
+        home_config / 'bapctools' / 'config.yaml',
+    ]:
```

Comment:
We could even add `tools_root/'config'/'bapctools.yaml'` in here for global defaults, but for now the small dictionary in `config.py` is simpler and no need to read even more files during startup

Reply:
Anyway I think I'd like to add `contest.yaml` to this as well (but maybe a separate PR, since that likely adds a whole lot more changes), and `../../` variants to allow repo-root config from problem directories.

Really it would be cleaner to parse these after `cd`ing to the contest dir, but that probably introduces other problems

Reply:
As we discussed on Slack, I think that `contest.yaml` should stay separate from the command-line args, to keep it clearer where certain values are coming from.

I've tried separating the `cd`ing to the contest dir before parsing the personal config file in TODO. This commit is on a separate branch, so it doesn't interfere with this PR, so we can move the discussion elsewhere. Splitting off a `change_directory()` function from the `get_problems()` function already makes things slightly cleaner, but:
- Moving `change_directory()` outside `run_parsed_args` means we'll have to duplicate this between `main()` and `test(args)`
- This will also require changes in the behaviour of `new_{contest,problem}`, as the `--contest` and `--problem` flags are not valid for these commands